circle.yml

```diff
@@ -1,21 +1,17 @@
 machine:
   environment:
     DMD_VERSION: 2.071.1
-    PATH: "${HOME}/dmd2/linux/bin64:${PATH}"
-    LD_LIBRARY_PATH: "${HOME}/dmd2/linux/lib64:${LD_LIBRARY_PATH}"
-    SELF_COMPILE: 2
     SELF_DMD_TEST_COVERAGE: 1
-    DC: dmd
-    DMD: dmd
 dependencies:
   pre:
     - sudo apt-get update; sudo apt-get install g++-multilib
   override:
-    - curl -fsSL --retry 3 "http://downloads.dlang.org/releases/2.x/${DMD_VERSION}/dmd.${DMD_VERSION}.linux.tar.xz" | tar -C ~ -Jxf -
+    - for i in {0..4}; do curl -fsS -A "$CURL_USER_AGENT" --max-time 5 https://dlang.org/install.sh -O && break || [ $i -ge 4 ] || sleep $((1 << $i)); done
+    - source $(bash install.sh dmd-$DMD_VERSION --activate)
```

Comment:
Unfortunately this won't work for CircleCI has all commands are run in separate shells.